resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesSslConfigurationProvider.scala

```diff
@@ -0,0 +1,203 @@
+        sslEnvs += new EnvVarBuilder()
+          .withName(ENV_SUBMISSION_KEYSTORE_PASSWORD_FILE)
+          .withValue(s"$sslSecretsDirectory/$SUBMISSION_SSL_KEYSTORE_PASSWORD_SECRET_NAME")
+          .build()
+      })
+      driverSubmitSslOptions.keyPassword.foreach(password => {
+        val passwordBase64 = BaseEncoding.base64().encode(password.getBytes(Charsets.UTF_8))
+        sslSecretsMap += (SUBMISSION_SSL_KEY_PASSWORD_SECRET_NAME -> passwordBase64)
+        sslEnvs += new EnvVarBuilder()
+          .withName(ENV_SUBMISSION_KEYSTORE_KEY_PASSWORD_FILE)
+          .withValue(s"$sslSecretsDirectory/$SUBMISSION_SSL_KEY_PASSWORD_SECRET_NAME")
+          .build()
+      })
+      driverSubmitSslOptions.keyStoreType.foreach(storeType => {
+        sslEnvs += new EnvVarBuilder()
+          .withName(ENV_SUBMISSION_KEYSTORE_TYPE)
+          .withValue(storeType)
+          .build()
+      })
+      sslEnvs += new EnvVarBuilder()
+        .withName(ENV_SUBMISSION_USE_SSL)
+        .withValue("true")
+        .build()
+      val sslVolume = new VolumeBuilder()
+        .withName(SUBMISSION_SSL_SECRETS_VOLUME_NAME)
+        .withNewSecret()
+        .withSecretName(sslSecretsName)
+        .endSecret()
+        .build()
+      val sslVolumeMount = new VolumeMountBuilder()
+        .withName(SUBMISSION_SSL_SECRETS_VOLUME_NAME)
+        .withReadOnly(true)
+        .withMountPath(sslSecretsDirectory)
+        .build()
+      val sslSecrets = kubernetesClient.secrets().createNew()
+        .withNewMetadata()
+        .withName(sslSecretsName)
+        .endMetadata()
+        .withData(sslSecretsMap.asJava)
+        .withType("Opaque")
+        .done()
+      kubernetesResourceCleaner.registerOrUpdateResource(sslSecrets)
+      secrets += sslSecrets
+      val (driverSubmitClientTrustManager, driverSubmitClientSslContext) =
+        buildSslConnectionConfiguration(driverSubmitSslOptions)
+      KubernetesSslConfiguration(
+        driverSubmitSslOptions,
+        isKeyStoreLocalFile,
+        sslEnvs.toArray,
+        Array(sslVolume),
+        Array(sslVolumeMount),
+        secrets.toArray,
+        driverSubmitClientTrustManager,
+        driverSubmitClientSslContext)
+    } else {
+      KubernetesSslConfiguration(
+        driverSubmitSslOptions,
+        isKeyStoreLocalFile,
+        Array[EnvVar](),
+        Array[Volume](),
+        Array[VolumeMount](),
+        Array[Secret](),
+        None,
+        SSLContext.getDefault)
+    }
+  }
+
+  private def parseDriverSubmitSslOptions(): (SSLOptions, Boolean) = {
+    val maybeKeyStore = sparkConf.get(KUBERNETES_DRIVER_SUBMIT_KEYSTORE)
+    val resolvedSparkConf = sparkConf.clone()
+    val (isLocalKeyStore, resolvedKeyStore) = maybeKeyStore.map(keyStore => {
+      val keyStoreURI = Utils.resolveURI(keyStore)
+      val isProvidedKeyStoreLocal = keyStoreURI.getScheme match {
+        case "file" | null => true
+        case "container" => false
```

Comment:
I think container isn't supposed to work anymore after this change https://github.com/apache-spark-on-k8s/spark/pull/107/files#diff-b5527f236b253e0d9f5db5164bdb43e9R79